### PR TITLE
Add 3D support to `blob_dog` and `blob_log`

### PR DESCRIPTION
**Pull request by jni**
## Description
Replaces #2114.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [x] Unit tests

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/feature/corner.py [hunk 1]
```diff
@@ -182,11 +182,36 @@ def hessian_matrix(image, sigma=1, mode='constant', cval=0, order=None):
 
     return H_elems
 
-def hessian_matrix_det(image, sigma=1):
+
+def _hessian_matrix_image(H_elems):
+    """Convert the upper-diagonal elements of the Hessian matrix to a matrix.
```
Comment (stefanv) [inline skimage/feature/corner.py]: Double check one line description

**Comment by jni** [comment]
Now complete with deprecations, new tests, and a bonus rebase!

**Comment by jni** [comment]
Note: I'm not confident about the eigenvalue sorting. Specifically, should they be sorted by absolute value or by magnitude? Reading a bit about how to use the Hessian eigendecomposition, I'm starting to think that it's magnitude that is critical. @scikit-image/core can anyone clarify? Note that it would be easily fixable with a call to np.sort.

**Comment by Phlya** [comment]
Hi, just curious whether this is getting merged any time soon? Would love to try it!

**Comment by jni** [comment]
@Phlya do you know the answer to my question about eigenvalue sorting? =)

**Comment by Phlya** [comment]
@jni nope, sorry, wish I could help! :)

**Pull request merged**